lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -54,7 +78,7 @@ def serializable_hash_for_single_resource(serializer, options)
           primary_data = primary_data_for(serializer, options)
           relationships = relationships_for(serializer)
           included = included_for(serializer)
-          hash = { data: primary_data }
+          hash[:data] = primary_data # FIXME: should be an [], not a {}
```

Comment:
More than legal, it is actually the right behavior when targeting a single resource.